src/clientSideScene/segments.ts

```diff
@@ -823,8 +828,37 @@ function createLengthIndicator({
   lengthIndicatorText.innerText = roundOff(length).toString()
   const lengthIndicatorWrapper = document.createElement('div')
 
+  // Double click workflow
+  lengthIndicatorWrapper.ondblclick = () => {
+    const selection = lengthIndicatorGroup.parent?.userData.selection
+    if (!selection) {
+      console.error('Unable to dimension segment when clicking the label.')
+      return
+    }
+    sceneInfra.modelingSend({
+      type: 'Set selection',
+      data: {
+        selectionType: 'singleCodeCursor',
+        selection: selection.graphSelections[0],
+      },
+    })
+
+    // Command Bar
+    editorManager.commandBarSend({
+      type: 'Find and select command',
+      data: {
+        name: 'Constrain length',
+        groupId: 'modeling',
+        argDefaultValues: {
+          selection,
+        },
+      },
+    })
+  }
+
   // Style the elements
   lengthIndicatorWrapper.style.position = 'absolute'
+  lengthIndicatorWrapper.style.pointerEvents = 'auto'
```

Comment:
overwrite the child component since the parent overlay is `pointerEvents:none`. This will change the mouse icon to indicate the user can click the label.